### PR TITLE
Consider sensitivity when evaluating module outputs

### PR DESCRIPTION
**Pull request by pselle**
This change "marks" values related to outputs that have Sensitive set to true. Sensitivity will follow outputs used elsewhere in configuration:

```terraform

# main.tf

module "foo" {
  source = "./mod"
}

resource "test_instance" "x" {
  some_attribute = module.mod.a
}

# mod/main.tf

output "a" {
  value = "secret"
  sensitive = true"
}
```

Locations where the sensitive output is used will be redacted in CLI output:

```sh
Terraform will perform the following actions:

  # test_instance.x will be created
  + resource "test_instance" "x" {
      + some_attribute    = (sensitive)
    }

```

### CODE REVIEW

**Review by alisdair — DISMISSED** [review]
This looks great to me!

I think this is the first time that we're joining up the idea of sensitive outputs and sensitive variables. Is there some documentation we should update accordingly?

**Comment by pselle** [comment]
@alisdair Good call-out on docs -- I'll add the docs in this PR and re-request review. Something in the output/sensitive documentation would be good. FWIW, this change is related to sensitive outputs using the marking strategy we use for sensitive variables, but it's actually separate from variables!

File: website/docs/configuration/outputs.html.md [hunk 1]
```diff
@@ -98,10 +98,53 @@ output "db_password" {
 }
 ```
 
-Setting an output value in the root module as sensitive prevents Terraform
-from showing its value in the list of outputs at the end of `terraform apply`.
-It might still be shown in the CLI output for other reasons, like if the
```
Comment (pselle) [inline website/docs/configuration/outputs.html.md:103]: The spirit of this section, referencing <0.14 behavior, has been migrated to the **Note** below

**Review by alisdair — APPROVED** [review]
Doc changes look great!

**Pull request merged**